packages/workbox-precaching/controllers/PrecacheController.mjs

```diff
@@ -176,4 +177,90 @@ export default class PrecacheController {
 
     return true;
   }
+
+  /**
+   * Compare the URL's and determines which assets are no longer required
+   * in the cache.
+   *
+   * This should be called in the service worker activate event.
+   *
+   * @return {Promise<object>} Resolves with an object containing details
+   * of the deleted cache requests and precache revision details.
+   */
+  async cleanup() {
+    const expectedCacheUrls = [];
+    this._entriesToCacheMap.forEach((entry) => {
+      expectedCacheUrls.push(entry._cacheRequest.url);
+    });
+
+    const results = await Promise.all([
```

Comment:
Minor nit: personally I'd use destructuring here to keep it concise but add more descriptive names:

```js
const [cacheRequests, revisionDetails] = Promise.all(...)
```